### PR TITLE
Sf3 compatibility

### PR DESCRIPTION
**Pull request by wasbaiti**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 vendor
 composer.lock
+.idea
```
Comment (malarzm) [inline .gitignore]: Please keep IDE specific entries in your global `.gitignore` file

File: .travis.yml [hunk 2]
```diff
@@ -16,17 +16,31 @@ before_script:
 
 notifications:
   email:
-    - malarzm@gmail.com
+    - walid.asbaiti@gmail.com
```
Comment (malarzm) [inline .travis.yml]: You can remove `notifications` section all together, I did so in https://github.com/codete/FormGeneratorBundle/commit/0cf28fbfe902721543be4510b8dc5125b94d5384

File: .travis.yml [hunk 3]
```diff
@@ -16,17 +16,31 @@ before_script:
...
+#    - php: 5.5
+#      env: SYMFONY_VERSION=2.4.*
+#    - php: 5.5
+#      env: SYMFONY_VERSION=2.5.*
+#    - php: 5.5
+#      env: SYMFONY_VERSION=dev-master
+#    - php: 5.5
+#      env: SYMFONY_VERSION=2.6.*
+#    - php: 5.5
+#      env: SYMFONY_VERSION=2.7.*
```
Comment (malarzm) [inline .travis.yml]: `2.7` should be uncommented and the rest removed

File: FormGenerator.php [hunk 4]
```diff
@@ -93,14 +97,19 @@ public function addFormViewProvider(FormViewProviderInterface $provider, $priori
      */
     public function createFormBuilder($model, $form = 'default', $context = array())
     {
-        $fb = $this->formFactory->createBuilder('form', $model);
+        if ($this->sfMajorVersion < 3) {
```
Comment (malarzm) [inline FormGenerator.php]: Actually we could use your `FieldTypeMapper` instead of doubling checks here

File: composer.json [hunk 5]
```diff
@@ -9,13 +9,13 @@
         {"name": "Maciej Malarz", "email": "malarzm@gmail.com"}
     ],
     "require": {
-        "php": ">=5.3.2",
         "doctrine/annotations": "~1.0",
-        "symfony/form": "~2.1",
-        "symfony/framework-bundle": "~2.3"
+        "symfony/form": ">=2.7",
+        "symfony/framework-bundle": ">=2.7",
+        "symfony/symfony": "3.0.*"
```
Comment (malarzm) [inline composer.json]: Why are we introducing this dependency?

**Comment by malarzm** [comment]
@wasbaiti thanks for the PR! I left some remarks, if you could also rebase and squash we should be good to go :+1:

**Comment by wasbaiti** [comment]
@malarzm Thanks for your remarks.
Everything seems OK to me.

**Comment by malarzm** [comment]
Thanks @wasbaiti \o/

**Pull request merged**